tests/test_quantum_distance.py

```diff
@@ -69,30 +109,50 @@ def test_value_error_is_raised_if_vectors_dont_have_norm_1(
         raised when computing the quantum distance if one or both of
         them have norm different from 1.
         """
-        for i in range (self.n_sizes):
-            for j in range(args.n_samples -1):
-                vector_1 = self.vectors_that_dont_need_padding[i][j]
-                vector_2 = self.vectors_that_dont_need_padding[i][j + 1]
-                with self.assertRaises(ValueError):
-                    quantum_distance = qd(
-                        vector_1,vector_2
-                    ).compute_quantum_distance()
+        for j in range(self.n_samples -1):
+            vector_1 = self.vectors_that_dont_need_padding[j]
+            vector_2 = self.vectors_that_dont_need_padding[j + 1]
+            with self.assertRaises(ValueError):
+                quantum_distance = qd(
+                    vector_1,vector_2
+                ).compute_quantum_distance()
+
+    def test_value_error_is_raised_if_vectors_are_not_powers_of_two(
+        self
+    )-> None:
+        """
+        For two vectors with same length, test that a ValueError is
+        raised when computing the quantum distance if one or both of
+        them have norm different from 1.
+        """
+        for j in range(self.n_samples -1):
+            vector_1 = self.vectors_that_need_padding[j]
+            vector_1 = normalise_vector(vector_1)
```

Comment:
You generally do not want to be using your own library to make operations when you're testing for smth.
If this test fails, it could be either because my `normalise_vector` function is broken or because my `compute_quantum_distance` one is not throwing the error => I don't know why the test fails => this test is not useful.

If you need to normalise stuff, use a sturdy lib like numpy to do that.

In general test results shoul not depend on other functions than the one they're testing.

Reply:
Completely agree with what you say.  

I was looking up very quickly some libraries but I could'nt  find any that does exactly what we want. However you can see the functions, they are super simple: 

-normalise_vector is just one line doing : vector/np.linalg.norm(vector), so it's actually using numpy in a very simple way.
-pad_vector_with_zeros pads with zeros until the vector length is a power of two. No function in numpy does directly that.

If you want I can spend more time looking for libraries that has functions doing exactly what we want. but that of course will take more time...